CHANGELOG.md

```diff
@@ -16,6 +16,8 @@ These are the section headers that we use:
 
 ## [Unreleased]()
 
+## [1.24.0](https://github.com/argilla-io/argilla-server/releases/tag/v1.24.0)
+
```

Comment:
Should we add something explaining the change related with moving `server` code to a different repository?

Reply:
The correct format for notes using GitHub Markdown is here: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts